### PR TITLE
v2 Qualification Test Updates

### PR DESCRIPTION
**Pull request by andersonjake1988**
## Overview
IN WORK:
qual tests

Notes:

### CODE REVIEW

**Comment by andersonjake1988** [comment]
In the case of NA values for GroupID the data pipeline breaks down at the Transform_Rate step, which has a stop clause for that situation. I have incorporated logic in the Input_Rate step to filter out cases of NA GroupID and informing the user how many cases there were when processing the yaml

**Comment by lauramaxwell** [comment]
sounds good, @andersonjake1988. Can you make sure this branch is up to date with `dev` so that we are testing the most recent versions of the functions, and any edits to the codebase won't lead to conflicts when we try to get this merged in?

**Comment by andersonjake1988** [comment]
# NOTES:
- not sure if there is a reason, but I noticed the yamls for kri's and cou's 6, 7, and 12 have a nMinDenominator of 3 when everything else has 30.

**Review by lauramaxwell — CHANGES_REQUESTED** [review]
looking good! a few questions and edits and we should be gtg soon!

File: R/Input_Rate.R [hunk 1]
```diff
@@ -46,123 +46,123 @@
...
-#'   dfNumerator = clindata::rawplus_ae,
-#'   dfDenominator = clindata::rawplus_dm,
-#'   strSubjectCol = "subjid",
-#'   strGroupCol = "siteid",
-#'   strGroupLevel = "Site",
-#'   strNumeratorMethod = "Count",
-#'   strDenominatorMethod = "Sum",
-#'   strDenominatorCol = "timeontreatment"
-#' )
+#'     dfSubjects = clindata::rawplus_dm,
```
[the remark this answers is not in the document]
Comment (andersonjake1988) [reply R/Input_Rate.R:49]: I this added logic into the end to check for NA `GroupID` 
`if(any(is.na(dfInput$GroupID))){
        cli_alert_warning(glue::glue("{sum(is.na(dfInput$GroupID))} cases of NA's in GroupID, cases are removed in output"))
        dfInput <- dfInput %>%
            filter(!is.na(.data$GroupID))
    }`

File: inst/longitudinal_demo/LongitudinalReport.R [hunk 2]
```diff
@@ -0,0 +1,115 @@
+devtools::load_all()
```
[the remark this answers is not in the document]
Comment (andersonjake1988) [reply inst/longitudinal_demo/LongitudinalReport.R:1]: sorry it slipped through the cracks when I was merging dev, I will delete it

File: inst/qualification/qualification_specs.csv [hunk 3]
```diff
@@ -1,89 +1,11 @@
...
-23,5,"T23_1, T23_2, T23_3, T23_4, T23_5, T23_6, T23_7, T23_8, T23_9, T23_10, T23_11, T23_12",PD_Assess_Binary,"Given appropriate input data, flag values can be correctly assigned to records that meet flagging criteria, including custom thresholding.",Medium,High,Approved,,Approved,
-23,6,"T23_1, T23_2, T23_4, T23_5, T23_7, T23_8",PD_Assess_Binary,"Given an appropriate subset of input data, a Protocol Deviation Assessment can be done.",Medium,Medium,Approved,,Approved,
-23,7,"T23_10, T23_11, T23_12",PD_Assess_Binary,"Given appropriate input data, flag values are correctly assigned as NA for sites with low enrollment.",Medium,Medium,Approved,,Approved,
-24,1,"T24_1, T24_2, T24_3, T24_4",PD_Map_Raw_Binary,"Given appropriate metadata, the raw data for a Protocol Deviation Assessment can be correctly mapped to an analysis-ready input dataset.",Medium,High,Approved,Good.,Approved,
+1,1,T1_1,mapping.yaml,"lData is correctly mapped for processing using ""mapping.yaml"" in conjunction with `MakeWorkflowList()` and `RunWorkflow`",High,High,Awaiting Approval,,In progress,
```
[the remark this answers is not in the document]
Comment (andersonjake1988) [reply inst/qualification/qualification_specs.csv:2]: good catch I will update it

File: R/Input_Rate.R (hunk 1 above)
[the remark this answers is not in the document]
Comment (lauramaxwell) [reply R/Input_Rate.R:49]: great- thanks!

**Comment by lauramaxwell** [comment]
These qual tests all pass, and are appropriate, but since each file contains more than one test, the qualification report does not render properly. Since tests are run in an `imap`, there isn't a way to separate the test files, but I want to be able to account for each individual test. I know there some tweaks i need to make to the Qualification.Rmd, but see screenshot below for what i'm talking about - 24 tests and 311 tests pass... @jwildfire 

image

**Comment by jwildfire** [comment]
> These qual tests all pass, and are appropriate, but since each file contains more than one test, the qualification report does not render properly. Since tests are run in an `imap`, there isn't a way to separate the test files, but I want to be able to account for each individual test. I know there are some tweaks i need to make to the Qualification.Rmd, but see screenshot below for what i'm talking about - 24 tests and 311 tests pass... @jwildfire

Agreed. We'll want these split out into multiple line items in the report.

**Comment by lauramaxwell** [comment]
qualification report is fixed! will merge in shortly :)

**Comment by lauramaxwell** [comment]
QualificationWorkflow.Rmd still needs a bit of work. hoping to be able to get it done quickly

**Review by lauramaxwell — APPROVED** [review]

**Pull request merged**